### PR TITLE
fix(api-client): import curl query parameters

### PR DESCRIPTION
**Pull request by antlio**
this pr fixes query parameters importation form curl following latest changed introduced in #3267.

**after**

https://github.com/user-attachments/assets/b14a0428-dfcd-461d-b575-3a4efa2eaeb4

### CODE REVIEW

**Review by amritk — DISMISSED** [review]
Looks good but mind adding tests for them?

**Review by marclave — APPROVED** [review]

**Review by hanspagel — COMMENTED** [review]
So cool! We could decode the values, too.

```bash
curl 'https://void.scalar.com/foo?person=Marc%20(CEO)' \
  -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7' \
  -H 'accept-language: en-DE,en;q=0.9,de-DE;q=0.8,de;q=0.7,en-GB;q=0.6,en-US;q=0.5' \
  -H 'cookie: vector_visitor_id=null' \
  -H 'dnt: 1' \
  -H 'priority: u=0, i' \
  -H 'sec-ch-ua: "Google Chrome";v="129", "Not=A?Brand";v="8", "Chromium";v="129"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: document' \
  -H 'sec-fetch-mode: navigate' \
  -H 'sec-fetch-site: none' \
  -H 'sec-fetch-user: ?1' \
  -H 'upgrade-insecure-requests: 1' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36'
```

Screenshot 2024-10-15 at 13 25 41

I don’t know whether we’d be able to easily add support for arrays, too:

```
curl 'https://void.scalar.com/foo?items=1&items=2' \
  -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7' \
  -H 'accept-language: en-DE,en;q=0.9,de-DE;q=0.8,de;q=0.7,en-GB;q=0.6,en-US;q=0.5' \
  -H 'cache-control: max-age=0' \
  -H 'cookie: vector_visitor_id=null' \
  -H 'dnt: 1' \
  -H 'priority: u=0, i' \
  -H 'sec-ch-ua: "Google Chrome";v="129", "Not=A?Brand";v="8", "Chromium";v="129"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: document' \
  -H 'sec-fetch-mode: navigate' \
  -H 'sec-fetch-site: none' \
  -H 'sec-fetch-user: ?1' \
  -H 'upgrade-insecure-requests: 1' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36'
```

**Comment by hanspagel** [comment]
CI is actually green, it’s just the dependabot config from main that’s failing.

File: packages/api-client/src/libs/parse-curl.ts [hunk 1]
```diff
@@ -25,6 +26,24 @@ export function parseCurlCommand(curlCommand: string) {
 
   while (arg) {
     if (typeof arg === 'object' && 'op' in arg) {
+      if (arg.op === '&') {
```
Comment (amritk) [inline packages/api-client/src/libs/parse-curl.ts:29]: @antlio what does this do? If we remove it the test for query params still works. Also it seems to be runing the type for query params it should be an array not a record

Comment (antlio) [reply packages/api-client/src/libs/parse-curl.ts:29]: initially to identify query params, i might have added some updates that made this useless. updated it accordingly in #4384 tysm!

**Pull request merged**